### PR TITLE
feat: IP connection window inside main menu

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description (*)
Re-introducing IP connection to the MainMenu scene. This is triggered in the same way cheats are in-game (slash on Windows or four finger touch on mobile). Inspired by the LobbyPopup panel architecture, however for simplicity, `IPUIMediator` doesn't inject to DI.

Adjustments:
* `FinalizeScopeConstruction()` is invoked inside ClientMainMenuState regardless of authentication result. This seems necessary for DI to work elsewhere in the game.
* Added an `IsAuthenticated` check on queries for lobbies. This drastically limits warning messages in the console.

### Issue Number(s) (*)

Fixes issue(s): [MTT-2213](https://jira.unity3d.com/browse/MTT-2213)
### Manual testing scenarios

1. Build or create ParrelSync clone of project.
2. On both apps, press slash or 4 finger touch to bring up IP connection window. Connect as you would beforehand with IP addresses.
### Questions or comments
This PR doesn't remove the IP connection option that is presently there for lobby connections. Can easily add that to this PR.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by SamuelBellomo** [comment]
I don't see any changes to the IP flows in lobby? Since lobby direct IP connection can't work at the moment outside of localhost, we might want to change that flow?

**Comment by fernando-cortez** [comment]
> I don't see any changes to the IP flows in lobby? Since lobby direct IP connection can't work at the moment outside of localhost, we might want to change that flow?

Yup I remarked on that in the comments section. I'll get to that in this PR then!

File: Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:146]: I'm guessing this is waiting on LP's popup? which is waiting on your approval? :P

File: Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs
Comment (SamuelBellomo) [inline Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:186]: don't we already have sanitization logic somewhere else?

File: Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs
Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:146]: Correct! I suppose that PR is a dependency for this one.

File: Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs
Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:186]: Yes so that was moved to the new Lobby scripts from PopupPanel IIRC. So I'd just remove that from the lobby scripts and keep that here.

File: Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs
Comment (LPLafontaineB) [reply Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:146]: This shouldn't be necessary after the other pr, the connectFinished callback is removed and the connect status is published inside ClientGameNetPortal instead.

Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:146]: popup merged

**Review by LPLafontaineB — APPROVED** [review]

File: Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs
Comment (SamuelBellomo) [reply Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:166]: let me know when this is done and I'll approve

Comment (fernando-cortez) [reply Assets/BossRoom/Scripts/Client/UI/IPUIMediator.cs:166]: Good to test now.

**Review by SamuelBellomo — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**
